### ISSUE TITLE
pipe not working with very basic configuration

### ISSUE THREAD

**Issue by mb888** [mikebrady/shairport-sync#715]
I'm trying with no luck to get a very basic configuration to star with pipes. It will run without the pipes but with the "pipe" output _backend it fails to start every time. What am I missing here, it's not "rocket science"?

Here's the basic config:

general = {
  name = "test183";
  output_backend = "pipe";
};

pipe = {
     name = “/srv/media/iphone”;      
 };

I've made sure it was compiled with pipes using th shairport-sync -v command and tested the pipe by itself and it works fine. I just can't figure out why it won't allow shairport-sync to start.

**Comment by mikebrady** [comment]
Thanks for the post. There may be some information in the log that will help, but at a guess, it sounds like a permissions error. The user that shairport-sync is running as must have write permission to create the pipe and to write to it. Also, remember that if shairport-sync is running as a service, it will run as user `shairport-sync`.

**Comment by mb888** [comment]
Checked all permissions and they seem to be open for user shairport-sync for writes. That said, I did get it to start with pipes manually from the command line with "shairport-sync -o pipe -- /srv/music/iphone" but  it absolutely will not start my pipe via the config file.

**Comment by mikebrady** [comment]
Thanks for the report, and apologies for the delay. Let me check this out.

**Comment by mikebrady** [comment]
I can confirm that it works with a Raspberry Pi 3B+.
I used your settings above and I had to create the directory `media` within `srv`. I made its owner and group `shairport-sync`: 
```
pi@raspberrypi:/srv $ ls -al
total 12
drwxr-xr-x  3 root           root           4096 Jul  5 18:20 .
drwxr-xr-x 22 root           root           4096 Jun 22 21:06 ..
drwxr-xr-x  2 shairport-sync shairport-sync 4096 Jul  5 18:23 media
pi@raspberrypi:/srv $ 
```
Then I started the Shairport Sync service using `$ sudo systemctl start shairport-sync` and found that the `iphone` pipe had been created:
```
pi@raspberrypi:/srv $ cd media/
pi@raspberrypi:/srv/media $ ls -al
total 8
drwxr-xr-x 2 shairport-sync shairport-sync 4096 Jul  5 18:23 .
drwxr-xr-x 3 root           root           4096 Jul  5 18:20 ..
prw-r--r-- 1 shairport-sync shairport-sync    0 Jul  5 18:41 iphone
pi@raspberrypi:/srv/media $ 
```
I was able to play from it using:
```
$ cat /srv/media/iphone | aplay -f cd
```
without difficulty. Subsequently I `enabled` Shairport Sync to start as a service on boot:
```
$ sudo systemctl enable shairport-sync
```
and I can confirm that it starts, creates the pipe and one can `aplay` from it.

**Comment by mb888** [comment]
Mike,
Thanks for your undying support! After pulling what ever hair out I have left I found the stupid error that was causing the issue. The plain text editor I was using that's built into my sftp app was adding hidden formatting to the config file. I wouldn't have noticed it but I got lazy and remoted in from my iPad and used nano to edit and there the hidden code reared its ugly head. Once I cleaned it up it worked great!

Thanks again

**Comment by mb888** [comment]
Mike,
Thanks for your undying support! After pulling what ever hair out I have left I found the stupid error that was causing the issue. The plain text editor I was using that's built into my sftp app was adding hidden formatting to the config file. I wouldn't have noticed it but I got lazy and remoted in from my iPad and used nano to edit and there the hidden code reared its ugly head. Once I cleaned it up it worked great!

Thanks again
